src/python/pants/engine/legacy/graph.py

```diff
@@ -89,9 +90,14 @@ def _index(self, roots):
     # Index the ProductGraph.
     for node, state in roots.items():
       if type(state) is not Return:
-        trace = '\n'.join(self._scheduler.trace())
-        raise AddressLookupError(
-            'Build graph construction failed for {}:\n{}'.format(node, trace))
+        # NB: ResolveError means that a target was not found, which is a common user facing error.
+        #     So, we propagate that error's message.
```

Comment:
I'll add a note. The message is propagated, but the trace repr's it right now, so it is a bit obscured.

A stopgap is exactly what I was going for. This improves the behavior as it is now, and doesn't change the underlying structure much.

Where this gets tricky is in references because I'd like to be able to emulate the v1 behavior where we say where the missing address came from if it's being used by a BUILD file somewhere. Right now, that's tricky.